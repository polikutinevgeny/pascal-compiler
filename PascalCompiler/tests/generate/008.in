var a, b: Integer;
begin
    b := 2;
    a := 5;
    a := a * b + b;
    write(a);
end.