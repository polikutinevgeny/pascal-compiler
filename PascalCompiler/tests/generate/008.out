.686p
.mmx
.xmm
include \masm32\include\masm32rt.inc
.code
main PROC
Push Ebp
Mov Ebp, Esp
Sub Esp, 8
Push 2
Pop Eax
Mov dword ptr [Ebp-8], Eax
Push 5
Pop Eax
Mov dword ptr [Ebp-4], Eax
Push dword ptr [Ebp-4]
Push dword ptr [Ebp-8]
Pop Eax
Pop Ebx
Imul Eax, Ebx
Push Eax
Push dword ptr [Ebp-8]
Pop Eax
Pop Ebx
Add Eax, Ebx
Push Eax
Pop Eax
Mov dword ptr [Ebp-4], Eax
Push dword ptr [Ebp-4]
Pop Eax
printf("%d", Eax)
Mov Esp, Ebp
Pop Ebp
EXIT
main ENDP
END main
